app/main.py

```diff
@@ -2,24 +2,81 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int,
+                 max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: object, name: str) -> None:
+        self.private_name = "_" + name
+
+    def __get__(self, instance: object, owner: object) -> bool:
+        value = getattr(instance, self.private_name)
+        return self.min_amount <= value <= self.max_amount
+
+    def __set__(self, instance: object, value: int | float) -> None:
+        setattr(instance, self.private_name, value)
```

Comment:
The `__set__` method in the `IntegerRange` class should validate the `value` against `min_amount` and `max_amount` before setting it. This ensures that the value being set is within the allowed range.